pkg/printers/internalversion/printers.go

```diff
@@ -1409,11 +1410,31 @@ func printStatefulSet(obj *apps.StatefulSet, options printers.GenerateOptions) (
 	desiredReplicas := obj.Spec.Replicas
 	readyReplicas := obj.Status.ReadyReplicas
 	createTime := translateTimestampSince(obj.CreationTimestamp)
-	row.Cells = append(row.Cells, obj.Name, fmt.Sprintf("%d/%d", int64(readyReplicas), int64(desiredReplicas)), createTime)
+
+	status := "Unknown"
+	if obj.Status.ObservedGeneration != nil &&
```

Comment:
It might be more clear to use a switch, rather than four if statements? This would also different statuses to have a higher priority and remove duplicate condition checks. It might also be a bit cleaner if this is also split out into a separate function.

```go
status := statefulSetStatus(obj)
```

```go
func statefulSetStatus(obj *apps.StatefulSet) string {
	switch {
	case obj.ObjectMeta.DeletionTimestamp != nil:
		return "Terminating"
	case obj.Status.CurrentRevision != obj.Status.UpdateRevision:
		return "Updating"
	case obj.Status.Replicas != obj.Spec.Replicas:
		return "Scaling"
	case obj.Status.ObservedGeneration != nil && *obj.Status.ObservedGeneration == obj.Generation:
		return "Ready"
	}
	return "Unknown"
}
```

Reply:
@uhthomas thanks for your review. done.